code/controllers/subsystem/movement/movement_types.dm

```diff
@@ -142,6 +148,10 @@
 		return
 	direction = dir
 
+/datum/move_loop/move/compare_loops(datum/move_loop/loop_type, priority, flags, extra_info, delay, timeout, dir)
+	if(..() && direction == dir)
+		return TRUE
+
```

Comment:
then turn this into somethin like this
this is honestly more up to personal taste, could ignore me if you'd like
```suggestion
	if(direction != dir)
		return FALSE
	return ..()
```

Reply:
i can see why you would want this here trough mainly we would want to first call the parent proc so we can avoid having to do all those logical comparisons for some of the bigger if's if they don't matter either way now of course we can just do something like `. = ..()
if(. && direction == dir)
	return
return FALSE`
cause technically it would have to be true from the very beginning and only can become false afterwards to have any effect.

Reply:
if you think this would be a better structure then yeah i can apply it

Reply:
honestly what you have now is fine, I thought about it, and really any one argument has the same chance of being the same or different as any other, so it doesn't really make a difference

Reply:
it's not like the early returns would make the logic more clear, since this code is almost boilerplate

Reply:
Ok then lets keep it like that.